lib/models/wallet_connect.dart

```diff
@@ -0,0 +1,25 @@
+/// WalletConnect Methods
+abstract class WcMethods {
+  /// Get the client to request a list of all accounts in wallet
+  static const wRequestAccounts = "wallet_requestAccounts";
+
+  /// Send qubic from an account to a destination address
+  static const wSendQubic = "sendQubic";
+
+  /// Send assets from an account to a destination address
+  static const wSendAsset = "sendAsset";
+}
+
+abstract class WcEvents {
+  /// The list of accounts has changed
+  static const accountsChanged = "accountsChanged";
+
+  /// The amount of tokens in an account has changed
+  static const tokenAmountChanged = "tokenAmountChanged";
+
+  /// The amount of qubics in an account has changed
+  static const amountChanged = "amountChanged";
+
+  /// A generic result of method invocation
+  static const methodResult = "methodResult";
+}
```

Comment:
@Qubic-Hub  no clear for me what "methodResults" is needed for.
I've seen other implementations supporting 'custom methods' and none has such an event.

Reply:
afaiu the responses are sent to the DApp using await connector.respondSessionRequest(response);
Please have a look.

Reply:
@sallymoc
We are not using respondSessionRequest since it requires the request.id and registerRequestHandler does not provide it. There is a way to get it but it's hackish: (  final id = web3Wallet.pendingRequests.getAll().first; // by https://alpha-docs.walletconnect.com/web3wallet/wallet-usage?platform=flutter#responding-to-session-requests and may lead to mismatches - ex. if a new session request is received before replying then replies get garbled)

We can either send events as replies (and use a container event as a reply - this is what its currently implemented) , or use respondSessionRequest with the above hackish fix, the change should take a few hours. 

Please advise on how you would like to proceed

Reply:
walletConnect.on('call_request', (payload) {
  final requestId = payload['id'];
  final method = payload['method'];
  final params = payload['params'];

  // Process the request based on the method
  print('Received request ID: $requestId');
  print('Method: $method');
  print('Params: $params');

  // Respond or reject the request based on user action
});



I have the feeling we are reinventing the wheel. 
How is done in the WC wallet flutter demo?